tests/unit/test_sonata_config.py

```diff
@@ -149,6 +149,7 @@ def test_parse_modifications(create_tmp_simulation_config_file):
     from neurodamus.core.configuration import SimConfig
 
     SimConfig.init(create_tmp_simulation_config_file, {})
+    assert list(SimConfig.modifications.keys()) == ["no_SK_E2", "applyTTX"]  # order preserved
```

Comment:
are you sure that the order is preserved or that the order is required to be preserved?
```suggestion
    assert set(SimConfig.modifications.keys()) == {"no_SK_E2", "applyTTX"}
```

Reply:
Yes, the order is required to be preserved. The modifications should be done in order as in the simulation config file in case of multiple modifications on the same cells.

Reply:
yes but in this case you cannot have a dict for SimConfig.modifications. it is not granted that it preserves order. Any cycling on that may spark errors

Reply:
It is preserved from python 3.7.

Reply:
Right. We don't have the deletion case. I will resolved this conversation and go ahead to merge this PR.